### PR TITLE
Update to Iceberg 1.2.0

### PR DESCRIPTION
**Pull request by nastra**
This also requires the changes from https://github.com/tabular-io/iceberg-rest-image/pull/7

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -57,6 +59,11 @@ services:
     environment:
       - MINIO_ROOT_USER=admin
       - MINIO_ROOT_PASSWORD=password
+      - MINIO_DOMAIN=minio
```
Comment (nastra) [inline docker-compose.yml:62]: this is required for Minio to enable virtual-host-style access to the bucket (which is the default in the new AWS SDK version

Comment (nastra) [reply docker-compose.yml:62]: all of the networking changes are basically only to make virtual-host-style access to the bucket properly work

Comment (nastra) [reply docker-compose.yml:62]: the alternative would be to configure path-style access (which is deprecated) similar to https://github.com/tabular-io/docker-spark-iceberg/pull/70

File: spark/Dockerfile [hunk 2]
```diff
@@ -42,35 +42,32 @@ RUN curl https://github.com/SpencerPark/IJava/releases/download/v1.3.0/ijava-1.3
...
 RUN curl https://tabular-repository-public.s3.amazonaws.com/releases/io/tabular/tabular-client-runtime/${TABULAR_VERSION}/tabular-client-runtime-${TABULAR_VERSION}.jar -Lo /opt/spark/jars/tabular-client-runtime.jar
 
 # Download Java AWS SDK
-RUN curl https://repo1.maven.org/maven2/software/amazon/awssdk/bundle/2.17.257/bundle-2.17.257.jar -Lo /opt/spark/jars/bundle-2.17.257.jar
+ENV AWSSDK_VERSION=2.20.18
+RUN curl https://repo1.maven.org/maven2/software/amazon/awssdk/bundle/${AWSSDK_VERSION}/bundle-${AWSSDK_VERSION}.jar -Lo /opt/spark/jars/bundle-${AWSSDK_VERSION}.jar
 
 # Download URL connection client required for S3FileIO
-RUN curl https://repo1.maven.org/maven2/software/amazon/awssdk/url-connection-client/2.17.257/url-connection-client-2.17.257.jar -Lo /opt/spark/jars/url-connection-client-2.17.257.jar
+RUN curl https://repo1.maven.org/maven2/software/amazon/awssdk/url-connection-client/${AWSSDK_VERSION}/url-connection-client-${AWSSDK_VERSION}.jar -Lo /opt/spark/jars/url-connection-client-${AWSSDK_VERSION}.jar
```
Comment (danielcweeks) [inline spark/Dockerfile:70]: Did the change to switch over to the apache client make it into 1.2?  Just wondering if we still need this.

**Review by danielcweeks — APPROVED** [review]

File: spark/Dockerfile (hunk 2 above)
Comment (nastra) [reply spark/Dockerfile:70]: https://github.com/apache/iceberg/commit/053028ceb6d3bc04c223a521600b053235bf344a didn't make it into 1.2 unfortunately, so we still need this

**Pull request merged**